File: projects/m1/027-the-collatz-conjecture/solutions/enricozoboli/solution_the_collatz_conjecture.py

while True:
    n = int(input("Enter number: "))
    n_final = n
    if n_final > 0:
        while True:
            if n_final % 2 == 0:
                n_final //=  2
                print(n_final)
                if n_final == 1:
                    break
                else:
                    continue
            else:
                n_final = (n_final * 3) +1
                print(n_final)
                continue
    else:
        break

File: projects/m1/027-the-collatz-conjecture/solutions/valentinaBazzacchi/solution-the-collatz-conjecture.js
while (true) {  
   // The instruction within the first while must be repeated only when the sequence of numbers (increased in the second while) ending with 1. 
   let n = +prompt("Enter a positive integer");
   // Declare an array variable for collect the numbers with .push()
   let seq = [n];
   // Condition to exit the program
   if (n <= 0) break;
      // Ending condition: the sequence stops when "n" become 1 trought the following conditions and operations
      while (n != 1) {
      
         if (n % 2 == 0) { 
         n = Math.floor(n / 2);
         seq.push(n);

         } else {
         n = (n * 3) + 1;
         seq.push(n);
         }
      } // End first while 
/* 
Turn array into string with one spaces between each number. 
This instruction must be placed outside the second loop for display a complete sequence.   
*/
alert(seq.join(" "));  
   
} // End second while